src/installer/sources.py

```diff
@@ -2,14 +2,15 @@
 
 import os
 import posixpath
+import stat
 import zipfile
 from contextlib import contextmanager
 from typing import BinaryIO, Iterator, List, Tuple, cast
 
 import installer.records
 import installer.utils
 
-WheelContentElement = Tuple[Tuple[str, str, str], BinaryIO]
+WheelContentElement = Tuple[Tuple[str, str, str], BinaryIO, bool]
```

Comment:
YMMV, but when I add the third field, I tend to start wondering if it would be clearer with a dataclass rather than a tuple.

Reply:
Yea, I had the same thought too. :)

However, I want this to be decoupled such that the source can be implemented without a dependency on this package. I'm basically thinking that `wheel` would end up implementing this interface, in their public API, and then we'd have a nice amount of synergy.